cpp/src/arrow/type_traits.h

```diff
@@ -1069,6 +1069,13 @@ constexpr bool is_floating(Type::type type_id) {
   return false;
 }
 
+/// \brief Check for a physical floating point type
+///
+/// This predicate matches floating-point types, except half-float.
+constexpr bool is_physical_floating(Type::type type_id) {
```

Comment:
How about we use `switch cases` in this function and change `is_floating` to call `return is_physical_floating() and type_id == Type::HALF_FLOAT`?

Reply:
I don't know, I like to think of `is_floating` as the "primary" function (the one that defines the category).